demisto_sdk/commands/find_dependencies/find_dependencies.py

```diff
@@ -967,7 +992,59 @@ def _find_pack_dependencies(pack_id, id_set, verbose_file, exclude_ignored_depen
         return pack_dependencies
 
     @staticmethod
-    def build_dependency_graph(pack_id, id_set, verbose_file, exclude_ignored_dependencies=True):
+    def build_all_dependencies_graph(pack_ids: list,
+                                     id_set: dict,
+                                     verbose_file: VerboseFile,
+                                     exclude_ignored_dependencies: bool = True) -> nx.DiGraph:
+        """
+        Builds all level of dependencies and returns dependency graph for all packs
+
+        Args:
+            pack_ids (list): pack ids, currently pack folder names is in use.
+            id_set (dict): id set json.
+            verbose_file (VerboseFile): path to dependency explanations file.
+            exclude_ignored_dependencies (bool): Determines whether to include unsupported dependencies or not.
+
+        Returns:
+            DiGraph: all dependencies of given packs.
+        """
+        dependency_graph = nx.DiGraph()
+        for pack in pack_ids:
+            dependency_graph.add_node(pack, mandatory_for_packs=[])
+        for pack in dependency_graph.nodes():
+            dependencies = PackDependencies._find_pack_dependencies(
+                pack, id_set, verbose_file=verbose_file, exclude_ignored_dependencies=exclude_ignored_dependencies)
+            for dependency_name, is_mandatory in dependencies:
+                if dependency_name == pack:
+                    continue
+                if dependency_name not in dependency_graph:
+                    dependency_graph.add_node(dependency_name, mandatory_for_packs=[])
```

Comment:
when will this happen? is that purely for having a complete solution?

Reply:
In the pack_ids list there is a selection of lists with ignore list.
As far as I have seen `Base` pack is in it.

In such case - we don't want to calculate the dependencies for the things that are in the ignore list, but we do want to calculate them as dependencies.

In that case we need to add them as node to the graph.